src/pages/HomePage.tsx

```diff
@@ -1,60 +1,66 @@
+import { useSearchParams, Outlet } from 'react-router-dom';
+import { useGetCharactersListQuery } from '../redux/api/apiSlice';
 import ErrorBoundary from '../components/ErrorBoundary';
-import type { CharacterData } from '../models/interfaces';
 import CharacterCard from '../components/CharacterCard';
 import Pagination from '../components/Pagination';
 import Loader from '../components/Loader';
-import { Outlet } from 'react-router-dom';
+import { useState } from 'react';
+import SelectedItems from '../components/SelectedItems';
 import { useSelector } from 'react-redux';
 import { selectSelectedItems } from '../redux/slices/selectedItemsSlice';
-import SelectedItems from '../components/SelectedItems';
-
-interface HomePageProps {
-  isLoading: boolean;
-  isClichedErrorButton: boolean;
-  characters: CharacterData[];
-  errorMessage: string;
-  throwError: () => void;
-  currentPage: number;
-  setCurrentPage: (page: number) => void;
-  pages: number;
-}
 
-export default function HomePage({
-  isLoading,
-  isClichedErrorButton,
-  characters,
-  errorMessage,
-  throwError,
-  pages,
-  currentPage,
-  setCurrentPage,
-}: HomePageProps) {
+export default function HomePage() {
   const { selectedItems } = useSelector(selectSelectedItems);
+  const [isClichedErrorButton, setIsClichedErrorButton] = useState(false);
+  const [params, setParams] = useSearchParams();
+  const page = Number(params.get('page') || 1);
+  const name = params.get('name') || '';
+
+  const { data, isLoading, error, isError } = useGetCharactersListQuery({
+    page,
+    name,
+  });
+
+  const throwError = () => {
+    setIsClichedErrorButton(true);
+    console.error('Error: The Error boundary button was triggered');
+  };
+
+  const updatePage = (newPage: number) => {
+    setParams({ page: newPage.toString(), name });
+  };
+
+  const characters = data?.results || [];
+  const pages = data?.info?.pages || 0;
 
   return (
     <div className="m-auto px-0 py-4 justify-between align-top">
       <ErrorBoundary isClichedErrorButton={isClichedErrorButton}>
         <div className="m-auto mt-4 flex justify-between items-center">
           <button
-            className="bg-green-500 hover:bg-green-600 hover:text-white p-2 duration-300  py-2 px-4 text-black"
+            className="bg-green-500 hover:bg-green-600 hover:text-white p-2 duration-300  py-2 px-4"
             onClick={throwError}
           >
             ErrorBoundary
           </button>
           {pages > 0 && (
             <Pagination
-              currentPage={currentPage}
+              currentPage={page}
               pages={pages}
-              prevPage={currentPage > 1}
-              nextPage={currentPage < pages}
-              setCurrentPage={setCurrentPage}
+              prevPage={page > 1}
```

Comment:
It looks like `prevPage` and `nextPage` might be calculated inside of `Pagination` component.